file-formats/aifp/type/zif_aff_aifp_v1.intf.abap

```diff
@@ -0,0 +1,223 @@
+      message_class      TYPE c LENGTH 20,
+      "! <p class="shorttext">Message Number</p>
+      "! Message number
+      message_number     TYPE c LENGTH 3,
+      "! <p class="shorttext">Message Variable 1</p>
+      "! Message variable 1
+      message_variable_1 TYPE string,
+      "! <p class="shorttext">Message Variable 2</p>
+      "! Message variable 2
+      message_variable_2 TYPE string,
+      "! <p class="shorttext">Message Variable 3</p>
+      "! Message variable 3
+      message_variable_3 TYPE string,
+      "! <p class="shorttext">Message Variable 4</p>
+      "! Message variable 4
+      message_variable_4 TYPE string,
+    END OF ty_message.
+  TYPES:
+    "! <p class="shorttext">Field Check</p>
+    "! Field check
+    BEGIN OF ty_field_check,
+      "! <p class="shorttext">Field Check</p>
+      "! Field check
+      field_check          TYPE ty_simple_field_check,
+      "! <p class="shorttext">Field Check Operator</p>
+      "! Field Check Operator
+      field_check_operator TYPE zif_aff_types_v1=>ty_option,
+      "! <p class="shorttext">Field Check Pattern</p>
+      "! Pattern for Field Check
+      field_check_pattern  TYPE c LENGTH 60,
+    END OF ty_field_check.
+  TYPES:
+    "! <p class="shorttext">Database Check</p>
+    "! Database check
+    BEGIN OF ty_database_check,
+      "! <p class="shorttext">Table Name</p>
+      "! Table name
+      table_name              TYPE c LENGTH 45,
+      "! <p class="shorttext">Where Condition for Select Statement</p>
+      "! Where Condition for Select Statement
+      where_condition         TYPE c LENGTH 200,
+      "! <p class="shorttext">Database Check Type</p>
+      "! Database check type
+      database_check_type     TYPE ty_database_check_type,
+      "! <p class="shorttext">Field Name for Comparison</p>
+      "! Field Name for Comparison
+      field_name              TYPE c LENGTH 45,
+      "! <p class="shorttext">Database Check Operator</p>
+      "! Database check operator
+      database_check_operator TYPE zif_aff_types_v1=>ty_option,
+      "! <p class="shorttext">Database Check Pattern</p>
+      "! Pattern for database check
+      database_check_pattern  TYPE c LENGTH 60,
+    END OF ty_database_check.
+  TYPES:
+    "! <p class="shorttext">Class Check</p>
+    "! Class check
+    BEGIN OF ty_class_check,
+      "! <p class="shorttext">Check Class</p>
+      "! Check class
+      check_class TYPE zif_aff_types_v1=>ty_object_name_30,
+    END OF ty_class_check.
+  TYPES:
+    "! <p class="shorttext">Single Check</p>
+    "! Single check
+    BEGIN OF ty_single_check,
+      "! <p class="shorttext">Check Number</p>
+      "! Check number
+      "! $required
+      check_number      TYPE n LENGTH 3,
+      "! <p class="shorttext">Check Description</p>
+      "! Check description
+      check_description TYPE c LENGTH 120,
+      "! <p class="shorttext">Check Scenario</p>
+      "! Check scenario
+      check_scenario    TYPE ty_check_scenario,
```

Comment:
minor; up to you if you want to change it.
For these 3 fields you could remove the word "check", since it is clear from the structure title that it is referring to the check